xonsh/events.py

```diff
@@ -178,8 +187,22 @@ def discard(self, item):
         self._fired.discard(item)
         self._unfired.discard(item)
 
+    def _call(self, handler):
+        try:
+            handler(*self._pargs, **self._kwargs)
+        except Exception:
+            print_exception("Exception raised in event handler; ignored.")
+
     def fire(self, *pargs, **kwargs):
-        raise NotImplementedError("See #1550")
+        if self._hasfired:
+            return
```

Comment:
This seems like the event is only fireable only once, not just that each handle is callable only once. Should there be a reset method or some such so that the event can be fired more than once.  I don't have a compelling use case, so this is a non-blocking suggestion.